### PR TITLE
Update job config, add logging, improve materialization memory usage

### PR DESCRIPTION
**Pull request by mjurkus**
- Updated image pull policy
- Updated logs
- Accidentally included materialization memory performance PR from original repoa

### CODE REVIEW

**Review by danielsalvador — APPROVED** [review]

File: sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_engine.py [hunk 1]
```diff
@@ -273,7 +272,7 @@ def _print_pod_logs(self, job_id, feature_view, offset=0):
             label_selector=f"job-name={job_id}",
         ).items
         for i, pod in enumerate(pods_list):
-            logger.info(f"Logging output for {feature_view.name} pod {offset+i}")
+            logger.info(f"Logging output for {feature_view.name} pod {offset + i}")
```
Comment (KarolisKont) [inline sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_engine.py]: I think you need to fix the formating of ` + ` and use lazy formatting `%s` for all log messages.

Comment (KarolisKont) [reply sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_engine.py]: By fix, a mean use `# fmt: off` and `# fmt: on`

File: sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_dataflow.py [hunk 2]
```diff
@@ -31,14 +30,18 @@ def __init__(
...
         self.worker_index = worker_index
         self.paths = paths
+        self.mini_batch_size = int(
+            os.getenv("BYTEWAX_MINI_BATCH_SIZE", DEFAULT_BATCH_SIZE)
+        )
 
         self._run_dataflow()
 
     def process_path(self, path):
+        logger.info(f"Processing path {path}")
```
Comment (vstrimaitis) [inline sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_dataflow.py]: Use lazy formatting in logs (same applies in other places)
```suggestion
        logger.info(f"Processing path %s", path)
```

File: sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_engine.py [hunk 3]
```diff
@@ -355,7 +356,7 @@ def _create_job_definition(self, job_id, namespace, pods, env, index_offset=0):
             },
             {
                 "name": "BYTEWAX_REPLICAS",
-                "value": f"{pods}",
+                "value": "1",
```
Comment (vstrimaitis) [inline sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_engine.py:354]: Why?

**Review by vstrimaitis — COMMENTED** [review]
> Accidentally included materialization memory performance PR from original repoa

😁

File: sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_engine.py (hunk 3 above)
Comment (mjurkus) [reply sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_engine.py:354]: Was included in the original PR.

Comment (vstrimaitis) [reply sdk/python/feast/infra/materialization/contrib/bytewax/bytewax_materialization_engine.py:354]: Well alright I guess 😁

**Review by vstrimaitis — APPROVED** [review]

**Review by GytisBraz — APPROVED** [review]

**Pull request merged**